### PR TITLE
Migrate `cip` CLI command to cobra

### PR DESCRIPTION
**Pull request by justaugustus**
#### What type of PR is this?

/kind cleanup feature design

#### What this PR does / why we need it:

We use `cobra` for RelEng tooling, so this PR converts the image promotion to use it as well.

- Use cobra for `cip` command
- Fixup all references to `cip` commands
  - Update cip build target (`//:cip` --> `//cmd/cip:cip`)
  - Use cobra flags
  - Set `//cmd/cip:cip_lib` target visibility to public
  - Update `//:cip-auditor` `container_image` to use `cip audit` subcommand
  - test-e2e/cip: Drop klog flag in command invocation
  - Drop `make test-mac` target
- cip/cmd: Move auditor functionality into separate subcommand
- cip/cmd: Move promotion functionality into separate subcommand
- Update READMEs with new command invocations
- cmd/cip: Minor cleanup of flag descriptions
- cmd/cip: Prefer lowercased output formats (but strings.ToLower() anyway)
- cmd/cip: Add `cip version` command 

#### Does this PR introduce a user-facing change?

```release-note
- Use cobra for `cip` command
- Fixup all references to `cip` commands
  - Update cip build target (`//:cip` --> `//cmd/cip:cip`)
  - Use cobra flags
  - Set `//cmd/cip:cip_lib` target visibility to public
  - Update `//:cip-auditor` `container_image` to use `cip audit` subcommand
  - test-e2e/cip: Drop klog flag in command invocation
  - Drop `make test-mac` target
- cip/cmd: Move auditor functionality into separate subcommand
- cip/cmd: Move promotion functionality into separate subcommand
- cmd/cip: Prefer lowercased output formats (but strings.ToLower() anyway)
- cmd/cip: Add `cip version` command 
```

### CODE REVIEW

**Comment by justaugustus** [comment]
Ready for review.

/assign @hasheddan @saschagrunert @cpanato @puerco 
cc: @kubernetes-sigs/release-engineering

**Comment by justaugustus** [comment]
Flake:
/test pull-cip-auditor-e2e

**Review by xmudrii — COMMENTED** [review]
I've quickly gone through the PR, and overall it looks fine to me. I've left two tiny nits about potential improvements.

**Comment by justaugustus** [comment]
@xmudrii -- Same thoughts, but I was initially going to take these as follow-ups, as this PR has gotten way larger than I originally wanted/expected.

> Nit: should we have a dedicated `version` command instead of a flag (e.g. like krel)? The dedicated command approach might be a little bit more elegant and easier to implement because we wouldn't have to add logic for checking is the flag provided to each command.

Added in https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/282/commits/e2965d6f31542acfdd29e8c9cb521fd152cf2835, but I'll take cleanup of the old version commands as a follow-up as they interact with bazel and I'm not sure if modifying them will affect tests yet.

> Tiny nit: why we use uppercase here? Could we make this case-insensitive?

Fixed in https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/282/commits/b2de725eb98835d89c506565fc2ee495b754f822.

**Comment by justaugustus** [comment]
Flake:
/test pull-cip-auditor-e2e

**Comment by justaugustus** [comment]
(Successful GCB run: https://console.cloud.google.com/cloud-build/builds/871cfd6d-4a30-4281-97dc-5289ba098a2d?project=k8s-staging-artifact-promoter)

**Comment by justaugustus** [comment]
Flake:
/test pull-cip-auditor-e2e

File: cmd/cip/cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		rootOpts.dryRun,
+		"test run promotion without modifying any registry",
+	)
+
+	rootCmd.PersistentFlags().BoolVar(
+		&rootOpts.version,
+		"version",
+		rootOpts.version,
+		"print version",
+	)
```
[the remark this answers is not in the document]
Comment (justaugustus) [reply cmd/cip/cmd/root.go:84]: ref: https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/282#issuecomment-735351685

File: cmd/cip/cmd/run.go [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+				rii = sc.RemoveChildDigestEntries(rii)
+			}
+		}
+
+		var snapshot string
+		switch opts.outputFormat {
+		case "CSV":
+			snapshot = rii.ToCSV()
+		case "YAML":
+			snapshot = rii.ToYAML(reg.YamlMarshalingOpts{})
```
[the remark this answers is not in the document]
Comment (justaugustus) [reply cmd/cip/cmd/run.go]: ref: https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/282#issuecomment-735351685

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**